core/trino-main/src/main/java/io/trino/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -389,8 +389,18 @@ protected Node visitShowColumns(ShowColumns showColumns, Void context)
             if (!metadata.schemaExists(session, new CatalogSchemaName(tableName.getCatalogName(), tableName.getSchemaName()))) {
                 throw semanticException(SCHEMA_NOT_FOUND, showColumns, "Schema '%s' does not exist", tableName.getSchemaName());
             }
-            if (metadata.getView(session, tableName).isEmpty() &&
-                    metadata.getTableHandle(session, tableName).isEmpty()) {
+
+            Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
+            if (tableHandle.isEmpty()) {
+                throw semanticException(TABLE_NOT_FOUND, showColumns, "Table '%s' does not exist", tableName);
+            }
+            else {
+                // Queries against information_schema should never fail, so in case of DESC is there an issue in getting
+                // table metadata the error could be swallowed. To avoid this we call here getTableMetadata.
+                metadata.getTableMetadata(session, tableHandle.orElseThrow());
```

Comment:
Why can't we use `tableHandle.get` ?

Reply:
get() is deprecated.

> API Note:
The preferred alternative to this method is orElseThrow().

Reply:
Can we capture them in error prone compiler. Maybe in the future

Reply:
It is just  _preferred_.

Reply:
It's not deprecated and also one should not use `orElseThrow` blindly.
The difference between `get` and `orElseThrow` is in _tooling_ around this.
For example, IntelliJ warns you when you use `get` without proper checks, so one should still prefer to use `get` to get this benefit.
Else, using `orElseThrow` throws this benefit away.

Personally, I use `orElseThrow` when _I_ know it's safe by _my IDE_ does not.